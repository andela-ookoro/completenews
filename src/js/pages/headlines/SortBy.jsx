import React from 'react';
import PropTypes from 'prop-types';

// class to display sortBy
const SortBy = ({ data, onClick }) => (
  <button value={data} onClick={onClick}> {data} </button>
);
SortBy.propTypes = {
  data: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
export default SortBy;
