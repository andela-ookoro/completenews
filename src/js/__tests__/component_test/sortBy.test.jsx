import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import SortBy from '../../pages/headlines/SortBy';

const data = 'Top';
const onClick = jest.fn();
describe('rendering', () => {
  it('should render content as describe in the component', () => {
    const component = Renderer.create(
      <SortBy data={data} source={data} onClick={onClick} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  const component = shallow(
    <SortBy data={data} source={data} onClick={onClick} />,
  );

  it('should render a button with the sort option as text' +
      '(such as Top articles)', () => {
    const button = component.find('button');
    expect(button.text()).toEqual(`${data} articles`);
  });

  it('Clicking the button should call a function passed as props', () => {
    const button = component.find('button');
    button.simulate('click');
    expect(onClick).toBeCalled();
  });
});

