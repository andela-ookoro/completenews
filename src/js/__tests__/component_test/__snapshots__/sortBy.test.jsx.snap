// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render a button 1`] = `
<button
  onClick={[Function]}
  value="test"
>
  test
</button>
`;

exports[`Value of render button should be same as props 1`] = `
<button
  onClick={[Function]}
  value="Top"
>
  Top
</button>
`;
